### PR TITLE
#8 set up qulice and test coverage

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #8 I've set up qulice and test coverage, extended test for `GhThread` and added todo to continue increasing code coverage.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#27](https://github.com/g4s8/ghman/issues/27) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#27](https://github.com/g4s8/ghman/issues/27) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CJT4R32Q3/5d35527b-3e96-41f2-af41-4233d24d93bb); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/com/g4s8/ghman/user/GhThreadTest.java [hunk 1]
```diff
@@ -32,11 +34,48 @@
     @Test
     void parseId() {
         final String tid = "4";
-        MatcherAssert.assertThat(
+        new Assertion<>(
+            "Failed to get thread id from json",
```
Comment (victornoel) [inline src/test/java/com/g4s8/ghman/user/GhThreadTest.java]: @olenagerasimova the preferred wording for assertions is the positive one, so here it should be "must get thread id from json".Could you update this one and the other tests here?

File: pom.xml [hunk 2]
```diff
@@ -25,6 +25,9 @@
   <version>1.0-SNAPSHOT</version>
   <properties>
     <telegram.version>3.6</telegram.version>
+    <surefire.version>3.0.0-M1</surefire.version>
+    <junit.version>5.4.2</junit.version>
+    <cactoos.dir>${basedir}/src/main/java/org/cactoos</cactoos.dir>
```
Comment (victornoel) [inline pom.xml]: @olenagerasimova this is used nowhere

File: pom.xml [hunk 3]
```diff
@@ -25,6 +25,9 @@
   <version>1.0-SNAPSHOT</version>
   <properties>
     <telegram.version>3.6</telegram.version>
+    <surefire.version>3.0.0-M1</surefire.version>
```
Comment (victornoel) [inline pom.xml]: @olenagerasimova it was 3.0.0-M3, why switch back to M1?

File: pom.xml [hunk 4]
```diff
@@ -151,11 +149,6 @@
       <artifactId>javax.json-api</artifactId>
       <version>1.1.4</version>
     </dependency>
-    <dependency>
-      <groupId>org.glassfish</groupId>
-      <artifactId>javax.json</artifactId>
```
Comment (victornoel) [inline pom.xml]: @olenagerasimova why was this removed?

File: pom.xml [hunk 5]
```diff
@@ -261,9 +263,13 @@
           <plugin>
             <groupId>org.jacoco</groupId>
             <artifactId>jacoco-maven-plugin</artifactId>
-            <version>0.8.1</version>
             <configuration>
               <output>file</output>
+              <excludes>
+                <exclude>**/*.sketch</exclude>
```
Comment (victornoel) [inline pom.xml]: @olenagerasimova why exclude this? there are no `.sketch` file in this repository, is there?

File: pom.xml [hunk 6]
```diff
@@ -261,9 +263,13 @@
...
+                <exclude>**/org/cactoos/**/*</exclude>
+                <exclude>**/ghman/data/*</exclude>
```
Comment (victornoel) [inline pom.xml]: @olenagerasimova why is this excluded?

File: pom.xml [hunk 7]
```diff
@@ -272,54 +278,33 @@
...
                   <goal>check</goal>
                 </goals>
                 <configuration>
-                  <dataFile>${project.build.directory}/jacoco-aggregate.exec</dataFile>
                   <rules>
                     <rule>
                       <element>BUNDLE</element>
+                      <!--
+                        @todo #8:30min Increase code coverage by
+                         writing unit tests. All coverage ration minimums are
```
Comment (victornoel) [inline pom.xml]: @olenagerasimova ration -> ratio

File: pom.xml (hunk 4 above)
Comment (olenagerasimova) [reply pom.xml]: @victornoel It was a duplicate

File: pom.xml (hunk 6 above)
Comment (olenagerasimova) [reply pom.xml]: @victornoel it's database related package, I do not think there will be any unit tests for classes of this package. But actually it might be unfair to exclude it from coverage metrics. @g4s8 what do say? Should we exclude it or not?

**Comment by olenagerasimova** [comment]
@victornoel I've made correction and added some comments. Also I've removed puzzle for this ticket (forget to do it earlier) and unnecessary `JavadocMethodCheck` suppression, qulice now understands it's a test and does not complain.  

As for this 
> Will this be visible directly from the PR with codecov or something like that or is this something else to setup on top of this?

I am afraid I do not understand what are toking about.

File: pom.xml (hunk 6 above)
Comment (victornoel) [reply pom.xml]: @olenagerasimova I don't see why there would be a different treatment for this particular package. Let's wait for @g4s8 ruling on this anyway :)

**Comment by victornoel** [comment]
@olenagerasimova thanks

@g4s8 apart from https://github.com/g4s8/ghman/pull/27#discussion_r292584193, it should be alright.
Also maybe it is needed to setup something specific to get feedback about coverage in PRs? cf discussion above.

File: pom.xml (hunk 6 above)
Comment (g4s8) [reply pom.xml]: @victornoel @olenagerasimova I think we can test `data` package with integration tests and then add it back to coverage metrics, WDYT?

Comment (victornoel) [reply pom.xml]: @g4s8 @olenagerasimova I just don't understand why we would need to first exclude them.

Comment (olenagerasimova) [reply pom.xml]: @victornoel @g4s8 I just remove this line to save our time

**Comment by olenagerasimova** [comment]
@victornoel I've removed `data` package exclusion

**Comment by olenagerasimova** [comment]
@victornoel ping

**Comment by victornoel** [comment]
@g4s8 it's alright with me

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@victornoel[/z](https://www.0crat.com/u/victornoel) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by olenagerasimova** [comment]
@g4s8 there are problems with puzzles from #2. What do we do here?

**Comment by g4s8** [comment]
@olenagerasimova we're waiting for @samihibrahim to fix puzzles in #2

**Comment by olenagerasimova** [comment]
@g4s8 please, try to merge, I've pulled master

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#27](https://github.com/g4s8/ghman/issues/27) is not in the agenda of @victornoel[/z](https://www.0crat.com/u/victornoel), can't inspect

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ammaratef45** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)
